### PR TITLE
Feature#290

### PR DESCRIPTION
**Pull request by wsbrenk**
1. changes proposed in this pull request:
 
   - fixes issue #290 
 
 
2. changelog and release_notes ...

 - [x] have been updated
 - [ ] do not require update

3. [Optional] suggested person to review this PR 

@tychobrailleur could you please check if my ui changes in DownloadDialog are OK?
@akasolace had a long fight against git, concerning rebasing - I have no idea if you think that's OK

### CODE REVIEW

File: src/main/java/core/gui/comp/CheckBoxTree/CheckBoxTree.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import java.awt.*;
+import java.awt.event.*;
+import java.util.EventListener;
+import java.util.EventObject;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class CheckBoxTree extends JTree {
```
Comment (tychobrailleur) [inline src/main/java/core/gui/comp/CheckBoxTree/CheckBoxTree.java:15]: Hey @wsbrenk can you please give attribution to https://stackoverflow.com/a/21851201/289466 ?  SO code is [CC BY-SA](https://creativecommons.org/licenses/by-sa/4.0/)

Comment (wsbrenk) [reply src/main/java/core/gui/comp/CheckBoxTree/CheckBoxTree.java:18]: @tychobrailleur is that sufficient?

Comment (tychobrailleur) [reply src/main/java/core/gui/comp/CheckBoxTree/CheckBoxTree.java:18]: Yes, thanks a mil!

**Comment by akasolace** [comment]
@wsbrenk this is weired I was sure that I have done the latest artefacts after merging this PR but I don not see the new download box ... same for you ?
image

**Comment by wsbrenk** [comment]
@akasolace  that i can't understand

grafik

created new local branch develop which checked out upstream/master and got this:


grafik

**Comment by akasolace** [comment]
@wsbrenk 
my bad I made the artefacts before I merged your PR ....

**Pull request merged**